components/table/src/Table.js

```diff
@@ -8,157 +8,160 @@ import { jsx, useTheme } from '@westpac/core';
 // Utils
 // ==============================
 
-const TableWrapper = ({ isBordered, isResponsive, children, ...props }) => {
-	const { table, breakpoints } = useTheme();
+const TableWrapper = ({ bordered, responsive, wrappingStyles, ...props }) => {
+	const {
+		COLORS,
+		LAYOUT: { breakpoints },
+	} = useTheme();
 
 	const maxWidth = width => `@media (max-width: ${width}px)`;
 	const xsOnly = maxWidth(breakpoints.sm - 1);
 
-	const common = {
-		[xsOnly]: {
-			width: '100%',
-			marginBottom: '1.8rem',
-			overflowY: 'hidden',
-			overflowX: 'auto',
-			// -ms-overflow-style: '-ms-autohiding-scrollbar',
-			border: `${table.responsive.borderWidth} solid ${table.responsive.borderColor}`,
-			// -webkit-overflow-scrolling: 'touch',
-
-			'> table': {
-				marginBottom: 0,
-				border: isBordered && 0,
-
-				caption: {
-					padding: table.responsive.caption.padding,
-				},
-				'tbody, tfoot': {
-					'tr:last-child': {
-						'th, td': {
-							borderBottom: isBordered && 0,
+	return responsive ? (
+		<div
+			css={[
+				wrappingStyles,
+				{
+					[xsOnly]: {
+						width: '100%',
+						marginBottom: '1.125rem',
+						overflowY: 'hidden',
+						overflowX: 'auto',
+						// -ms-overflow-style: '-ms-autohiding-scrollbar',
+						border: `1px solid ${COLORS.border}`,
+						// -webkit-overflow-scrolling: 'touch',
+
+						'> table': {
+							marginBottom: 0,
+							border: bordered && 0,
+
+							caption: {
+								padding: '0.75rem',
+							},
+							'tbody, tfoot': {
+								'tr:last-child': {
+									'th, td': {
+										borderBottom: bordered && 0,
+									},
+								},
+							},
+							'th, td': {
+								whiteSpace: 'nowrap',
+
+								':first-of-type': {
+									borderLeft: bordered && 0,
+								},
+								':last-child': {
+									borderRight: bordered && 0,
+								},
+							},
 						},
 					},
 				},
-				'th, td': {
-					whiteSpace: 'nowrap',
-
-					':first-of-type': {
-						borderLeft: isBordered && 0,
-					},
-					':last-child': {
-						borderRight: isBordered && 0,
-					},
-				},
-			},
-		},
-	};
-
-	return isResponsive ? (
-		<div className="table-responsive" css={common}>
-			{children}
-		</div>
+			]}
+			{...props}
+		/>
 	) : (
-		children
+		props.children
```

Comment:
use `children` like this (`props.children`), instead as `children` via destructure assignment, so then we can simply parse children via props?

`<div {...props} />` instead of `<div {...props}>{children}</div>`

Am I following?

Reply:
Actually I think that's my code. I'm going cray